OpenRA.Mods.Common/Scripting/Global/ActorGlobal.cs

```diff
@@ -64,14 +64,55 @@ public Actor Create(string type, bool addToWorld, LuaTable initTable)
 			return a;
 		}
 
-		[Desc("Returns the build time (in ticks) of the requested unit type.")]
-		public int BuildTime(string type)
+		[Desc("Returns the build time (in ticks) of the requested unit type."
+			+ "An optional second value can be used to exactly specify the actor with the producing queue.")]
+		public int BuildTime(string type, string producer = null)
```

Comment:
This should accept the queue name instead of an actor.  There's no reason why an actor couldn't produce the same actor from two different queues with their own time modifiers (e.g. a combined war-factory / starport).